radicle-cli/src/commands/remote.rs

```diff
@@ -0,0 +1,125 @@
+//! Remote Command implementation
+#[path = "remote/add.rs"]
+pub mod add;
+#[path = "remote/list.rs"]
+pub mod list;
+#[path = "remote/rm.rs"]
+pub mod rm;
+
+use std::ffi::OsString;
+
+use anyhow::anyhow;
+
+use radicle::prelude::Did;
+
+use crate::terminal as term;
+use crate::terminal::args::{string, Error};
+use crate::terminal::{Args, Context, Help};
+
+pub const HELP: Help = Help {
+    name: "remote",
+    description: "Manage set of tracked repositories",
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+    rad remote
+    rad remote list
+    rad remote add <url> [--alias]
+    rad remote rm <alias>
+Options
+        --help                 Print help
+"#,
+};
+
+#[derive(Debug, Default, PartialEq, Eq)]
+pub enum OperationName {
+    Add,
+    Rm,
+    #[default]
+    List,
+}
+
+#[derive(Debug)]
+pub enum Operation {
+    Add { did: Did, alias: Option<String> },
+    Rm { alias: String },
+    List,
+}
+
+#[derive(Debug)]
+pub struct Options {
+    pub op: Operation,
+    pub verbose: bool,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut op: Option<OperationName> = None;
+        let mut did: Option<Did> = None;
+        let mut alias: Option<String> = None;
+        let mut verbose = false;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Long("verbose") | Short('v') => {
+                    verbose = true;
+                }
+                Long("alias") | Short('a') => {
+                    let value = parser.value()?;
+                    let value = string(&value);
+                    alias = Some(value);
+                }
+                Value(val) if op.is_none() => match val.to_string_lossy().as_ref() {
+                    "a" | "add" => op = Some(OperationName::Add),
+                    "l" | "list" => op = Some(OperationName::List),
+                    "r" | "rm" => op = Some(OperationName::Rm),
+                    unknown => anyhow::bail!("unknown operation '{}'", unknown),
+                },
+                Value(val) if op == Some(OperationName::Add) && did.is_none() => {
+                    did = Some(term::args::did(&val)?);
+                }
```

Comment:
We should do what `rad track` does and accept a NodeID or the NodeID in DID format.

```
                    } else if let Ok(did) = term::args::did(val) {
                        op = Some(Operation::TrackNode {
                            nid: did.into(),
                            alias: None,
                        });
                    } else if let Ok(nid) = term::args::nid(val) {
                        op = Some(Operation::TrackNode { nid, alias: None });
                    }
```

Reply:
Mh interesting, I will look at it thanks!

Reply:
Done in https://github.com/radicle-dev/heartwood/pull/524/commits/d2ee404a4186722ce2c8f8aef202c2ab1dad4062 I think this solution simplifies a little bit the code?

Reply:
Concept is good, lets keep it as a separate commit, or maybe even a separate PR to update other commands which require a 'Node'.